In [141]:
#import dependencies
import pandas as pd
import numpy as np

#load csv file
csv_path = "Resources/purchase_data.csv"

#read csv file
purchase_data = pd.read_csv(csv_path)

In [183]:
#display purchase_data df
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Ranges  
0  Extraction, Quickblade Of Trembling Hands   3.53      20-24  
1                          Frenzied Scimitar   1.56        40+  
2                               Final Critic   4.88      20-24  
3                                Blindscythe   3.27      20-24  
4                                       Fury   1.44      20-24

In [201]:
#Display the total number of unique players
unique_names = purchase_data["SN"].unique()
total_players_df = len(unique_names)

unique_players = pd.DataFrame({"Total Players": [total]})
unique_players

Total Players
0            576

In [202]:
#Purchasing Analysis (Total)

#calculate unique items
unique_items = purchase_data["Item ID"].unique()
total_items = len(unique_items)

#calculate average price
average_price = purchase_data["Price"].mean()

#calculate total revenue
revenue_df = purchase_data["Price"].sum()

#calculate total amount of sales
sales_df = purchase_data["Purchase ID"].count()

In [203]:
#Create a summary data frame to hold the results
purchase_summary_df = pd.DataFrame({"Number of Unique Items": [total_items],
                                    "Average Price": [average_price],
                                    "Number of Purchases": [sales_df],
                                    "Total Revenue": [revenue_df]})

#Optional: give the displayed data cleaner formatting 
purchase_summary_df["Average Price"] = purchase_summary_df["Average Price"].map("${:.2f}".format)
purchase_summary_df["Total Revenue"] = purchase_summary_df["Total Revenue"].map("${:,.2f}".format)

#Display the PURCHASING ANALYSIS summary data frame
purchase_summary_df

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     179         $3.05                  780     $2,379.77

In [204]:
#Gender Demographics

#get total count of genders
gender_df = purchase_data.groupby(["Gender"])

#calculate unique names grouped by gender
gender_count = gender_df["SN"].nunique()

#calculate percentages
percentage_df = gender_count/total*100
percentage_df

#create new data frame for gender analysis
demographics_df = pd.DataFrame({"Total Count": gender_count,
                                "Percentage of Players": percentage_df})

#sort total count from highest to lowest
demographics_df = demographics_df.sort_values("Total Count", ascending=False)

#remove Gender line
demographics_df = demographics_df.rename_axis([""])

#clean data and add percent sign
demographics_df["Percentage of Players"] = demographics_df["Percentage of Players"].map("{:.2f}%".format)

#Display the summary data frame for GENDER DEMOGRAPHICS
demographics_df

Total Count Percentage of Players
                                                        
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

In [205]:
#Purchasing Analysis (Gender)

#Calculate purchase count
purchase_count_df = gender_df["Purchase ID"].count()

#calculate average purchase price
average_count_df = gender_df["Price"].mean()

#calculate average purchase total
purchase_total_df = gender_df["Price"].sum()
average_total_df = purchase_total_df/gender_count

In [206]:
#Create a summary data frame to hold the results for PURCHASING ANALYSIS
purchasing_analysis_df = pd.DataFrame({"Purchase Count": purchase_count,
                                       "Average Purchase Price": average_count_df,
                                       "Total Purchase Value": purchase_total_df,
                                       "Avg Total Purchase per Person": average_total_df})

#clean data
purchasing_analysis_df["Average Purchase Price"]= purchasing_analysis_df["Average Purchase Price"].map("${:.2f}".format)
purchasing_analysis_df["Total Purchase Value"]= purchasing_analysis_df["Total Purchase Value"].map("${:,.2f}".format)
purchasing_analysis_df["Avg Total Purchase per Person"]= purchasing_analysis_df["Avg Total Purchase per Person"].map("${:.2f}".format)

#display data frame for PURCHASING ANALYSIS
purchasing_analysis_df

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase per Person  
Gender                                                                    
Female                             $361.94                         $4.47  
Male                             $1,967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

In [207]:
#Age Demographics

#Determine maximum and minimum ages from original data frame
maximum_age = purchase_data["Age"].max()
minimum_age = purchase_data["Age"].min()

#Establish bins for ages
bins = [5, 9.9, 14.9, 19.9 , 24.9, 29.9, 34.9, 39.9, 99]
label_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]

#use pd.cut to put new bins into original data frame
purchase_data["Age Ranges"] = pd.cut(purchase_data["Age"], bins, labels = label_names)

#calculate count by age range
age_range_df = purchase_data.groupby(["Age Ranges"])

#calculate total unique count of players
age_total_df = age_range_df["SN"].nunique()

#calculate percentage of players
age_percentage_df = age_total_df/total*100

In [208]:
#create new data frame for AGE DEMOGRAPHICS
age_demographic_df = pd.DataFrame({"Total Count": age_total_df, 
                                   "Percentage of Players": age_percentage_df})

#clean data
age_demographic_df["Percentage of Players"]= age_demographic_df["Percentage of Players"].map("{:.2f}%".format)

#display AGE DEMOGRAPHICS data frame
age_demographic_df

Total Count Percentage of Players
Age Ranges                                   
<10                  17                 2.95%
10-14                22                 3.82%
15-19               107                18.58%
20-24               258                44.79%
25-29                77                13.37%
30-34                52                 9.03%
35-39                31                 5.38%
40+                  12                 2.08%

In [209]:
#Purchasing Analysis (Age)

#calculate purchase count by age ranges
purchase_count_age = age_range_df["Purchase ID"].count()

#calculate average purchase price by age range
average_purchase_df = age_range_df["Price"].mean()

#find total purchase value by age range
total_purchase_df = age_range_df["Price"].sum()

#find average total purchase per person by age range
average_total_purchase = total_purchase_df/age_total_df

In [210]:
#create new data frame for PURCHASING ANALYSIS (AGE)
purchase_analysis_age = pd.DataFrame({"Purchase Count": purchase_count_age, 
                                      "Average Purchase Price": average_purchase_df, 
                                      "Total Purchase Value": total_purchase_df, 
                                      "Avg Total per Person": average_total_purchase})

#clean data
purchase_analysis_age["Average Purchase Price"]= purchase_analysis_age["Average Purchase Price"].map("${:.2f}".format)
purchase_analysis_age["Total Purchase Value"]= purchase_analysis_age["Total Purchase Value"].map("${:,.2f}".format)
purchase_analysis_age["Avg Total per Person"]= purchase_analysis_age["Avg Total per Person"].map("${:.2f}".format)

#display PURCHASING ANALYSIS (AGE)
purchase_analysis_age

Purchase Count Average Purchase Price Total Purchase Value  \
Age Ranges                                                               
<10                     23                  $3.35               $77.13   
10-14                   28                  $2.96               $82.78   
15-19                  136                  $3.04              $412.89   
20-24                  365                  $3.05            $1,114.06   
25-29                  101                  $2.90              $293.00   
30-34                   73                  $2.93              $214.00   
35-39                   41                  $3.60              $147.67   
40+                     13                  $2.94               $38.24   

           Avg Total per Person  
Age Ranges                       
<10                       $4.54  
10-14                     $3.76  
15-19                     $3.86  
20-24                     $4.32  
25-29                     $3.81  
30-34                     $4.12  
35-39                     $4.76  
40+                       $3.19

In [211]:
#Top Spenders

#Use groupby to calculate top spenders
spenders_df = purchase_data.groupby(["SN"])

#calculate total purchases for each spender
spenders_purchases = spenders_df["Price"].sum()

#calculate purchase count per spender
spenders_count = spenders_df["Purchase ID"].count()

#calculate average purchase price
spenders_average = spenders_df["Price"].mean()

In [212]:
#create new data frame for TOP SPENDERS
all_spenders_df = pd.DataFrame({"Purchase Count": spenders_count, 
                                "Average Purchase Price": spenders_average, 
                                "Total Purchase Value": spenders_purchases})

#sort data 
top_spenders_df = all_spenders_df.sort_values(["Total Purchase Value"], ascending=False)

#clean data
top_spenders_df["Average Purchase Price"]= top_spenders_df["Average Purchase Price"].map("${:.2f}".format)
top_spenders_df["Total Purchase Value"]= top_spenders_df["Total Purchase Value"].map("${:.2f}".format)

#display TOP SPENDERS data frame
top_spenders_df.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

In [213]:
#Most Popular Items

#Retrieve the Item ID, Item Name, and Item Price columns
popular_items = purchase_data.groupby(["Item ID", "Item Name"])

#calculate purchase count of popular items
popular_count = popular_items["Purchase ID"].count()

#calculate total price by groups
popular_total = popular_items["Price"].sum()

#calculate item prices
popular_price = popular_total/popular_count

In [214]:
#create new data frame for MOST POPULAR ITEMS
popular_df = pd.DataFrame({"Purchase Count": popular_count, 
                           "Item Price": popular_price, 
                           "Total Purchase Value": popular_total})

#sort data
most_popular_df = popular_df.sort_values(["Purchase Count"], ascending=False)

#clean data
most_popular_df["Item Price"]= most_popular_df["Item Price"].map("${:.2f}".format)
most_popular_df["Total Purchase Value"]= most_popular_df["Total Purchase Value"].map("${:.2f}".format)

#display MOST POPULAR ITEMS data frame
most_popular_df.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

In [215]:
#Most Profitable Items

#sort previous data frame by purchase value
most_profitable_df = popular_df.sort_values(["Total Purchase Value"], ascending=False)

#clean data
most_profitable_df["Item Price"]= most_profitable_df["Item Price"].map("${:.2f}".format)
most_profitable_df["Total Purchase Value"]= most_profitable_df["Total Purchase Value"].map("${:.2f}".format)

#display MOST PROFITABLE ITEMS data frame
most_profitable_df.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80

Heroes of Pymoli Analysis

Purchasing Analysis (Total) - The game has revenued $2,379.77

Purchasing Analysis (Gender) - The game is more popular amongst male players over female and they are responsible for the most purchases

Purchasing Analysis (Age) - Players in the 20-24 age range make the most purchases and are responsible for nearly half of the total revenue

Top Spenders - The top spender spent less than $20 and only accounts for 5 out of a total 780 purchases.

Most Popular/Profitable Items - Final Critic is the most popular and most profitable item.